ee/tabby-webserver/src/service/github_repository_provider.rs

```diff
@@ -113,6 +117,83 @@ impl GithubRepositoryProviderService for GithubRepositoryProviderServiceImpl {
 
         Ok(urls)
     }
+
+    async fn refresh_repositories(&self) -> Result<()> {
+        let mut cached_repositories: HashSet<_> = self
+            .list_github_provided_repositories_by_provider(vec![], None, None, None, None)
+            .await?
+            .into_iter()
+            .map(|repo| repo.vendor_id)
+            .collect();
```

Comment:
handle each provider individually to allow better data concurrency (something like refresh_repositories(provider_id: ID)